libs/io/bbhnet/io/h5.py

```diff
@@ -32,7 +32,7 @@ def write_timeseries(
             f["out"] = y
 
         for key, value in datasets.items():
-            f[key] = value
+            f.create_dataset(name=key, data=str(value))

```

Comment:
I'm not sure you want to be writing the arrays as strings, is there a reason for this?

Reply:
rolled back to the original code in h5.py
Fixing the datasets object during testing resolved this